src/theory/strings/inference_manager.cpp

```diff
@@ -349,20 +305,72 @@ void InferenceManager::doPendingFacts()
 
 void InferenceManager::doPendingLemmas()
 {
-  if (!d_state.isInConflict())
+  if (d_state.isInConflict())
+  {
+    // just clear the pending vectors, nothing else to do
+    d_pendingLem.clear();
+    d_pendingReqPhase.clear();
+    return;
+  }
+  NodeManager* nm = NodeManager::currentNM();
+  for (unsigned i = 0, psize = d_pendingLem.size(); i < psize; i++)
```

Comment:
I don't see the index `i` being used. Should we maybe simply do `for (InferInfo& ii : d_pendingLem)`?

Reply:
Yes, I'll do that.

Reply:
Actually its used, we store the use_index based on it.